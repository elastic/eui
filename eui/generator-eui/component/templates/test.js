import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  test('is rendered', () => {
    const component = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
