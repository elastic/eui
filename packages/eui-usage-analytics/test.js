/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const Codeowners = require('codeowners');
const temp = new Codeowners("../../../kibana");

const { scan } = require('./scan');

const runScan = async () => {
  const scanResult = await scan();
  console.log(scanResult);
};

runScan();
