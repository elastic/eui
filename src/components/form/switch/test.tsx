import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
