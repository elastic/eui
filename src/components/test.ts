export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
