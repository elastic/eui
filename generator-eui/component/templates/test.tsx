/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  it('renders', () => {
    const { container } = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
