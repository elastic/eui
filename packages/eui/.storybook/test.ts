/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { queries, within, waitFor, fireEvent, expect } from '@storybook/test';
import * as dataTestSubjQueries from '../src/test/rtl/data_test_subj_queries';

/**
 * Custom Storybook within util with EUI query helpers
 * + additional chained async/waitFor component utils
 *
 * @see https://storybook.js.org/docs/writing-stories/play-function#writing-stories-with-the-play-function
 * @see https://testing-library.com/docs/dom-testing-library/api-within/
 */
const customWithin = (canvasElement: HTMLElement) => {
  const canvas = within<typeof queries & typeof dataTestSubjQueries>(
    canvasElement,
    { ...queries, ...dataTestSubjQueries }
  );

  return {
    ...canvas,

    /**
     * 1. Loki doesn't like userEvent, only fireEvent
     * 2. Storybook fires fireEvents too early (esp. on page load), so we add a waitFor
     */
    waitForAndClick: async (testSubject: string) => {
      await waitFor(() =>
        expect(canvas.getByTestSubject(testSubject)).toBeInTheDocument()
      );
      await fireEvent.click(canvas.getByTestSubject(testSubject));
    },

    waitForEuiPopoverVisible: async () =>
      await waitFor(() =>
        expect(canvasElement.querySelector('[data-popover-open]')).toBeVisible()
      ),
    waitForEuiPopoverHidden: async () =>
      await waitFor(() =>
        expect(
          canvasElement.querySelector('[data-popover-panel]')
        ).not.toBeInTheDocument()
      ),
  };
};

export { customWithin as within };
