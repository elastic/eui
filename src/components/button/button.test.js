import React from 'react';
import { render, mount } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import {
  EuiButton,
  COLORS,
  SIZES,
  ICON_SIDES,
} from './button';

describe('EuiButton', () => {
  test('is rendered', () => {
    const component = render(
      <EuiButton {...requiredProps}>
        Content
      </EuiButton>
    );

    expect(component)
      .toMatchSnapshot();
  });

  describe('props', () => {
    describe('fill', () => {
      it('is rendered', () => {
        const component = render(
          <EuiButton fill />
        );

        expect(component)
          .toMatchSnapshot();
      });
    });

    describe('isDisabled', () => {
      it('is rendered', () => {
        const component = render(
          <EuiButton isDisabled />
        );

        expect(component)
          .toMatchSnapshot();
      });
    });

    describe('isLoading', () => {
      it('is rendered', () => {
        const component = render(
          <EuiButton isLoading />
        );

        expect(component)
          .toMatchSnapshot();
      });
    });

    describe('iconType', () => {
      it('is rendered', () => {
        const component = render(
          <EuiButton iconType="user" />
        );

        expect(component)
          .toMatchSnapshot();
      });
    });

    describe('color', () => {
      COLORS.forEach(color => {
        test(`${color} is rendered`, () => {
          const component = render(
            <EuiButton color={color} />
          );

          expect(component)
            .toMatchSnapshot();
        });
      });
    });

    describe('size', () => {
      SIZES.forEach(size => {
        test(`${size} is rendered`, () => {
          const component = render(
            <EuiButton size={size} />
          );

          expect(component)
            .toMatchSnapshot();
        });
      });
    });

    describe('iconSide', () => {
      ICON_SIDES.forEach(iconSide => {
        test(`${iconSide} is rendered`, () => {
          const component = render(
            <EuiButton iconType="user" iconSide={iconSide}>
              Content
            </EuiButton>
          );

          expect(component)
            .toMatchSnapshot();
        });
      });
    });

    describe('href', () => {
      it('secures the rel attribute when the target is _blank', () => {
        const component = render(
          <EuiButton href="#" target="_blank" />
        );

        expect(component)
          .toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick and href', () => {
        const handler = jest.fn();
        const component = mount(
          <EuiButton href="#" onClick={handler} />
        );
        component.find('a').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });

      it('supports onClick as a button', () => {
        const handler = jest.fn();
        const component = mount(
          <EuiButton onClick={handler} />
        );
        component.find('button').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });
    });
  });
});
