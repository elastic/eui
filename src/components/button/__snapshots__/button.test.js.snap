// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiButton is rendered 1`] = `
<button
  aria-label="aria-label"
  class="euiButton euiButton--primary testClass1 testClass2"
  data-test-subj="test subject string"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButton props color danger is rendered 1`] = `
<button
  class="euiButton euiButton--danger"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props color ghost is rendered 1`] = `
<button
  class="euiButton euiButton--ghost"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props color primary is rendered 1`] = `
<button
  class="euiButton euiButton--primary"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props color secondary is rendered 1`] = `
<button
  class="euiButton euiButton--secondary"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props color warning is rendered 1`] = `
<button
  class="euiButton euiButton--warning"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props fill is rendered 1`] = `
<button
  class="euiButton euiButton--primary euiButton--fill"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props href secures the rel attribute when the target is _blank 1`] = `
<a
  class="euiButton euiButton--primary"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</a>
`;

exports[`EuiButton props iconSide left is rendered 1`] = `
<button
  class="euiButton euiButton--primary"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <svg
      aria-hidden="true"
      class="euiIcon euiButton__icon euiIcon--medium"
      height="18"
      viewBox="0 0 18 18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
      >
        <path
          d="M13.689 11.132c1.155 1.222 1.953 2.879 2.183 4.748a1.007 1.007 0 0 1-1 1.12H3.007a1.005 1.005 0 0 1-1-1.12c.23-1.87 1.028-3.526 2.183-4.748.247.228.505.442.782.633-1.038 1.069-1.765 2.55-1.972 4.237L14.872 16c-.204-1.686-.93-3.166-1.966-4.235a7.01 7.01 0 0 0 .783-.633zM8.939 1c1.9 0 3 2 4.38 2.633a2.483 2.483 0 0 1-1.88.867c-.298 0-.579-.06-.844-.157A3.726 3.726 0 0 1 7.69 5.75c-1.395 0-3.75.25-3.245-1.903C5.94 3 6.952 1 8.94 1z"
        />
        <path
          d="M8.94 2c2.205 0 4 1.794 4 4s-1.795 4-4 4c-2.207 0-4-1.794-4-4s1.793-4 4-4m0 9A5 5 0 1 0 8.937.999 5 5 0 0 0 8.94 11"
        />
      </g>
    </svg>
    <span
      class="euiButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButton props iconSide right is rendered 1`] = `
<button
  class="euiButton euiButton--primary euiButton--iconRight"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <svg
      aria-hidden="true"
      class="euiIcon euiButton__icon euiIcon--medium"
      height="18"
      viewBox="0 0 18 18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
      >
        <path
          d="M13.689 11.132c1.155 1.222 1.953 2.879 2.183 4.748a1.007 1.007 0 0 1-1 1.12H3.007a1.005 1.005 0 0 1-1-1.12c.23-1.87 1.028-3.526 2.183-4.748.247.228.505.442.782.633-1.038 1.069-1.765 2.55-1.972 4.237L14.872 16c-.204-1.686-.93-3.166-1.966-4.235a7.01 7.01 0 0 0 .783-.633zM8.939 1c1.9 0 3 2 4.38 2.633a2.483 2.483 0 0 1-1.88.867c-.298 0-.579-.06-.844-.157A3.726 3.726 0 0 1 7.69 5.75c-1.395 0-3.75.25-3.245-1.903C5.94 3 6.952 1 8.94 1z"
        />
        <path
          d="M8.94 2c2.205 0 4 1.794 4 4s-1.795 4-4 4c-2.207 0-4-1.794-4-4s1.793-4 4-4m0 9A5 5 0 1 0 8.937.999 5 5 0 0 0 8.94 11"
        />
      </g>
    </svg>
    <span
      class="euiButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`EuiButton props iconType is rendered 1`] = `
<button
  class="euiButton euiButton--primary"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <svg
      aria-hidden="true"
      class="euiIcon euiButton__icon euiIcon--medium"
      height="18"
      viewBox="0 0 18 18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
      >
        <path
          d="M13.689 11.132c1.155 1.222 1.953 2.879 2.183 4.748a1.007 1.007 0 0 1-1 1.12H3.007a1.005 1.005 0 0 1-1-1.12c.23-1.87 1.028-3.526 2.183-4.748.247.228.505.442.782.633-1.038 1.069-1.765 2.55-1.972 4.237L14.872 16c-.204-1.686-.93-3.166-1.966-4.235a7.01 7.01 0 0 0 .783-.633zM8.939 1c1.9 0 3 2 4.38 2.633a2.483 2.483 0 0 1-1.88.867c-.298 0-.579-.06-.844-.157A3.726 3.726 0 0 1 7.69 5.75c-1.395 0-3.75.25-3.245-1.903C5.94 3 6.952 1 8.94 1z"
        />
        <path
          d="M8.94 2c2.205 0 4 1.794 4 4s-1.795 4-4 4c-2.207 0-4-1.794-4-4s1.793-4 4-4m0 9A5 5 0 1 0 8.937.999 5 5 0 0 0 8.94 11"
        />
      </g>
    </svg>
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props isDisabled is rendered 1`] = `
<button
  class="euiButton euiButton--primary"
  disabled=""
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props isLoading is rendered 1`] = `
<button
  class="euiButton euiButton--primary"
  disabled=""
  type="button"
>
  <span
    class="euiButton__content"
  >
    <div
      class="euiLoadingSpinner euiLoadingSpinner--medium euiButton__spinner"
    />
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props size l is rendered 1`] = `
<button
  class="euiButton euiButton--primary euiButton--large"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;

exports[`EuiButton props size s is rendered 1`] = `
<button
  class="euiButton euiButton--primary euiButton--small"
  type="button"
>
  <span
    class="euiButton__content"
  >
    <span
      class="euiButton__text"
    />
  </span>
</button>
`;
